src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0) {
+      clearTimeout(timeoutId);
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0 || ev.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0) {
+      leftClick = true;
+    } else if (ev.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
+});
```

Comment:
The `thirdPromise` resolves when both left and right mouse clicks have been detected. However, it does not reset the `leftClick` and `rightClick` flags after resolving, which means subsequent clicks could cause unexpected behavior. Consider resetting these flags after the promise resolves.